client/fs_test.go

```diff
@@ -38,7 +39,18 @@ func withDaemon(t *testing.T, name string, fn func(ctl *Client)) {
 	repoPath, err := ioutil.TempDir("", "brig-client-repo")
 	require.Nil(t, err)
 
-	defer os.RemoveAll(repoPath)
+	defer func() {
+		// Somehow there is race condition between 
+		// srv.Close() from the defer at the very end 
+		// os.RemoveAll(repoPath).
+		// Theoretically, `go` should have closed server
+		// but in practice I see that repoPath is removed
+		// before server had a chance to close the DB
+		// and I see complains in log about DB.Close
+		// I introduce this time delay as a crude hack
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Hmm, not so convinced by this reasoning. Go defers are put on a stack and run in inverse order when leaving the function. Therefore the server close should come first... maybe something else deletes the directory before the close comes? This defer here should be executed last.  It could be also that a go routine in the server still runs when Close exists. Then this should be fixed instead.

Reply:
Yes, I agree that *theoretically* defers go in FILO order. But maybe badger Close() has some wierd logic and delayed execution.
Removing this Sleep() brings warning messages in the Test log.

The test itself is not failing but it is log clearly shows warning about unable to close DB. 

The test which you pointed out, passes for me. But I am quite sure that the issue is similar. The strange part we have not changed anything in DB related code. It is really a combination of HDD activity and the system load.

Reply:
Also the race condition would explain random fails with Travis test runs.

Reply:
Well, I would prefer to have the original race condition fixed instead of the workaround, but the fix is acceptable for the time being. :smirk: Approved.

Reply:
> Well, I would prefer to have the original race condition fixed instead of the workaround, but the fix is acceptable for the time being. smirk Approved.

I did a bit of digging, I suspect it is because `server.Close()` uses channels to communicate the quit/close. Thus we exit function before the intended event `db.Close()`. This is where my expertise unfortunately ends, I guess we need to wait for acknowledgement from the go routine.